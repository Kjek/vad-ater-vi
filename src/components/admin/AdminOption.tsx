import type { HTMLAttributes, MouseEventHandler } from 'react';
import InputButton from './Button';
import Text from '../atoms/Text';

interface AdminOptionProps extends HTMLAttributes<HTMLElement> {
  text?: string;
  buttonValue: string;
  onClick: MouseEventHandler<HTMLInputElement>;
}

const AdminOption = (props: AdminOptionProps) => {
  const { text, buttonValue, onClick } = props;
  return (
    <>
      <div className='flex w-full'>
        <Text className='flex-1 self-center' variant='h4'>
          {text}
        </Text>
        <InputButton value={buttonValue} onClick={onClick} />
      </div>
    </>
  );
};

export default AdminOption;
