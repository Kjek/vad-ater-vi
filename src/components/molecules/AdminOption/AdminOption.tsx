import InputButton from '@component/atoms/Button/Button';
import Text from '@component/atoms/Text/Text';
import type { HTMLAttributes, MouseEventHandler } from 'react';

interface AdminOptionProps extends HTMLAttributes<HTMLElement> {
  text?: string;
  buttonValue: string;
  onClick: MouseEventHandler<HTMLInputElement>;
}

const AdminOption = (props: AdminOptionProps) => {
  const { text, buttonValue, onClick } = props;
  return (
    <>
      <div className='flex w-full'>
        <Text className='flex-1 self-center' variant='h4'>
          {text}
        </Text>
        <InputButton value={buttonValue} onClick={onClick} />
      </div>
    </>
  );
};

export default AdminOption;
