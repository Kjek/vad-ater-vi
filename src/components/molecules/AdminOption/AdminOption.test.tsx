import { render } from '@testing-library/react';
import React from 'react';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import AdminOption from './AdminOption';

describe(AdminOption, () => {
  const onClick = vi.fn();
  let adminOption: HTMLElement;

  beforeEach(async () => {
    vi.resetAllMocks();
    adminOption = render(
      <AdminOption buttonValue='Press me!' onClick={onClick} />
    ).container;
  });

  it('renders correctly', () => {
    expect(adminOption).toBeInTheDocument();
    expect(adminOption).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = (
      <AdminOption buttonValue='Press me!' onClick={vi.fn()} />
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
