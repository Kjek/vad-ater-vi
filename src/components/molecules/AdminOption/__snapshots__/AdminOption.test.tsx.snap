// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdminOption > matches snapshot 1`] = `
<AdminOption
  buttonValue="Press me!"
  onClick={[MockFunction spy]}
/>
`;
